pkgs/shells/fish/default.nix

```diff
@@ -35,25 +35,55 @@ let
     #     source both, but source the more global configuration files earlier
     #     than the more local ones, so that more local configurations inherit
     #     from but override the more global locations.
-    
+
     if test -f /etc/fish/config.fish
       source /etc/fish/config.fish
     end
-    
+
     #                                                                             #
     ############### ↑ Nix hook for sourcing /etc/fish/config.fish ↑ ###############
   '';
 
   fishPreInitHooks = ''
-    # source nixos environment
+    # source nix environment
     # note that this is required:
     #   1. For all shells, not just login shells (mosh needs this as do some other command-line utilities)
     #   2. Before the shell is initialized, so that config snippets can find the commands they use on the PATH
-    builtin status --is-login
-    or test -z "$__fish_nixos_env_preinit_sourced" -a -z "$ETC_PROFILE_SOURCED" -a -z "$ETC_ZSHENV_SOURCED"
-    and test -f /etc/fish/nixos-env-preinit.fish
-    and source /etc/fish/nixos-env-preinit.fish
-    and set -gx __fish_nixos_env_preinit_sourced 1
+    if builtin status --is-login
+       or test -z "$__fish_nix_env_preinit_sourced" -a -z "$ETC_PROFILE_SOURCED" -a -z "$ETC_ZSHENV_SOURCED"
+
+      set -l nix_env_setup
+      # If we have a configuration provided by a NixOS or Nix-Darwin administrator, prefer it.
+      if test -f /etc/nix/support/set-environment.sh
+        set nix_env_setup /etc/nix/support/set-environment.sh
+      else
+        set -l common_prefix @NIX_STORE@/../var/nix/profiles/default/etc/profile.d
+        set -l script_selection nix.sh
+
+        string match -q (${coreutils}/bin/uname) Darwin
+        # Use `set -U nix_on_darwin_no_daemon` to tell Fish to use the single-user mode script instead
+        and not set -q nix_on_darwin_no_daemon
+        and set script_selection nix-daemon.sh
+
+        set nix_env_setup $common_prefix/$script_selection
+      end
+
+
+      # This happens before $__fish_datadir/config.fish sets fish_function_path, so it is currently
+      # unset. We set it and then completely erase it, leaving its configuration to $__fish_datadir/config.fish
```

Comment:
**Suggestion**: Change "completely erase" to "reset," since we aren't running `set -e` below. (I think this comment might have been copied over from the nix-darwin module?)

Reply:
So replying to any of these via email from my phone was a mistake. But yeah, erasing it in case it is empty may be unnecessary and at any rate the language of complete erasure is no longer accurate.